recipes/swift/recipe.yaml

```diff
@@ -0,0 +1,216 @@
+  - url: https://github.com/jpsim/Yams/archive/5.0.6.tar.gz
+    sha256: a81c6b93f5d26bae1b619b7f8babbfe7c8abacf95b85916961d488888df886fb
+    target_directory: Yams
+
+  # Additional components
+  - url: https://github.com/unicode-org/icu/archive/refs/heads/maint/maint-69.zip
+    sha256: 77939649ac5927a2900efae161ba7d038483e5890800766449a3d160bb7386f1
+    target_directory: icu
+  - url: https://github.com/swiftwasm/WasmKit/archive/refs/tags/0.0.3.tar.gz
+    sha256: a959a868dbacbd28d5ffc1b27fc7f2d56f076bc727bbdec5b0c315c489e13b58
+    target_directory: wasmkit
+  - url: https://github.com/WebAssembly/wasi-libc/archive/refs/tags/wasi-sdk-20.tar.gz
+    sha256: 0a1c09c8c1da62a1ba214254ff4c9db6b60979c00f648a5eae33831d6ee2840e
+    target_directory: wasi-libc
+  - url: https://github.com/apple/swift-format/archive/refs/heads/release/6.0.zip
+    sha256: 44b275ca2bc53aa373358e7a9ed84a7a130ab00e1119fc481a06a5b894e8f35f
+    target_directory: swift-format
+  - url: https://github.com/apple/swift-llvm-bindings/archive/refs/heads/swift/release/6.0.zip
+    sha256: b011be4c3e129b6cedf3ff59404ae6112b60020ec87c3892dd463609a8d7a94a
+    target_directory: swift-llvm-bindings
+  - url: https://github.com/apple/swift-foundation-icu/archive/refs/heads/release/6.0.zip
+    sha256: f52225aeb342de62db0cd3fa4fb56e5c05af6a3b121c2b71ddd02856f3cd6da5
+    target_directory: swift-foundation-icu
+  - url: https://github.com/apple/swift-foundation/archive/refs/heads/release/6.0.zip
+    sha256: 2094f4a826e146abc1142ad730c0e72762550b58ca567d39374e5756b58850c8
+    target_directory: swift-foundation
+  - url: https://github.com/madler/zlib/releases/download/v1.3/zlib13.zip
+    sha256: c561d09347f674f0d72692e7c75d9898919326c532aab7f8c07bb43b07efeb38
+    target_directory: zlib
+
+build:
+  skip:
+    - not linux
+  script: build.sh
+  number: 0
+
+requirements:
+  build:
+    - ${{ compiler('cxx') }}
+    - ${{ compiler('c') }}
+    - ${{ stdlib('c') }}
+    - clangxx
+    - clang
+    - cmake <4
+    - ninja
+    - git
+    - patch
+    - make
+    - bash
+    - binutils 2.43.*
+    # Python removed the `pipes` module in 3.13 which is used, so we need to use 3.12.
+    - python 3.12.*
+  host:
+    - glib
+    - glib-tools
+    - libgomp
+    - libxml2
+    - sqlite
+    - ncurses
+    - libedit
+    - swig
+    - rsync
+    - curl
+
+tests:
+  - script:
+    - swift --version
+    - swift --help
+
+about:
+  homepage: https://swift.org
+  summary: 'The Swift Programming Language'
+  description: |
+    Swift is a high-performance system programming language. It has a clean and modern syntax,
+    offers seamless access to existing C and Objective-C code and frameworks, and is memory-safe by default.
+
+    Swift is a complete and independent language that packages core features like flow control,
+    data structures, and functions, with high-level constructs like objects, protocols, closures, and generics.
+  license: Apache-2.0
+  license_file: LICENSE.txt
```

Comment:
This is going to be painful, but with all those source entries, we need to ensure that their licenses are also collected here, if relevant.

Reply:
Eventually, I would like to have individual package per source entry and have the final `swift` package depend (transitively) on all of them.

Would it ease license files' inclusion?